Detectors/MUON/MID/Simulation/test/testSimulation.cxx

```diff
@@ -0,0 +1,208 @@
+#include "MIDSimulation/DigitsMerger.h"
+#include "MIDSimulation/ChamberResponseParams.h"
+#include "MIDSimulation/Hit.h"
+#include "MIDClustering/PreClusterizer.h"
+#include "MIDClustering/Clusterizer.h"
+
+namespace o2
+{
+namespace mid
+{
+
+struct SIMUL {
+  static o2::mid::Mapping mapping;
+  static o2::mid::ChamberResponseParams params;
+  static o2::mid::GeometryTransformer geoTrans;
+  static o2::mid::Digitizer digitizer;
+  static o2::mid::DigitsMerger digitsMerger;
+  static o2::mid::PreClusterizer preClusterizer;
+  static o2::mid::Clusterizer clusterizer;
+
+  void setup()
+  {
+    clusterizer.init();
+  }
+};
+
+o2::mid::Mapping SIMUL::mapping;
+o2::mid::ChamberResponseParams SIMUL::params = createDefaultChamberResponseParams();
+o2::mid::GeometryTransformer SIMUL::geoTrans = createDefaultTransformer();
+o2::mid::Digitizer SIMUL::digitizer = createDefaultDigitizer();
+o2::mid::DigitsMerger SIMUL::digitsMerger;
+o2::mid::PreClusterizer SIMUL::preClusterizer;
+o2::mid::Clusterizer SIMUL::clusterizer;
+
+std::vector<Hit> generateHits(size_t nHits, int deId, const o2::mid::Mapping& mapping, const o2::mid::GeometryTransformer geoTrans)
+{
+  std::vector<Hit> hits;
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_real_distribution<double> distX(-127.5, 127.5);
+  std::uniform_real_distribution<double> distY(-40., 40.);
+  while (hits.size() < nHits) {
+    Point3D<float> point(distX(mt), distY(mt), 0.);
+    if (!mapping.stripByPosition(point.x(), point.y(), 0, deId, false).isValid()) {
+      continue;
+    }
+    Point3D<float> globalPoint = SIMUL::geoTrans.localToGlobal(deId, point);
+    hits.emplace_back(hits.size(), deId, globalPoint, globalPoint);
+  }
+  return hits;
+}
+
+bool checkLabel(const ColumnData& digit, MCLabel& label, std::string& errorMessage)
+{
+  std::stringstream ss;
+  ss << digit << "\n";
+  bool isOk = true;
+  if (label.getDEId() != digit.deId) {
+    isOk = false;
+    ss << "label deId " << label.getDEId() << " != " << digit.deId << "\n";
+  }
+  if (label.getColumnId() != digit.columnId) {
+    isOk = false;
+    ss << "label columnId " << label.getColumnId() << " != " << digit.columnId << "\n";
+  }
+  int firstStrip = label.getFirstStrip();
+  int lastStrip = label.getLastStrip();
+  int cathode = label.getCathode();
+  int nLines = (cathode == 0) ? 4 : 1;
+  for (int iline = 0; iline < nLines; ++iline) {
+    for (int istrip = 0; istrip < 16; ++istrip) {
+      int currStrip = MCLabel::getStrip(istrip, iline);
+      bool isFired = digit.isStripFired(istrip, cathode, iline);
+      if (isFired != (currStrip >= firstStrip && currStrip <= lastStrip)) {
+        isOk = false;
+        ss << "Cathode: " << cathode << "  firstStrip: " << firstStrip << "  lastStrip: " << lastStrip;
+        ss << "  line: " << iline << "  strip: " << istrip << "  fired: " << isFired << "\n";
+      }
+    }
+  }
+  errorMessage = ss.str();
+  return isOk;
+} // namespace mid
+
+std::vector<int> getDEList()
```

Comment:
would add a comment here to say why those 20 DEs where chosen out of the 72

Reply:
Ok